hello_world_on_chain/test/user.ts

```diff
@@ -1,79 +1,92 @@
+// noinspection DuplicatedCode
+
 import hre from "hardhat";
 import {loadFixture} from "@nomicfoundation/hardhat-toolbox/network-helpers";
 import {expect} from "chai";
 import {ethers} from "ethers";
+import fs from "fs";
+import path from "path";
+import {User} from "../typechain-types";
 
-const fs = require("fs")
-const path = require("path")
+type EVMLinkData = {
+    publicKey: string;
+    transitiveAttestation: string;
+};
 
-function loadEVMLinkData(jsonPath: string) {
+function loadEVMLinkData(jsonPath: string): EVMLinkData {
     try {
-        const dir = path.resolve( __dirname, jsonPath);
-        const file = fs.readFileSync(dir);
+        const dir: string = path.resolve( __dirname, jsonPath);
+        const file: string = fs.readFileSync(dir, "utf8");
 
-        const data = JSON.parse(file);
+        const data: any = JSON.parse(file);
 
-        const k = data.enclave_attested_application_public_key.claims.public_key.data
-        const pubKeyBytes = ethers.decodeBase64(k)
-        const publicKeyHex = Buffer.from(pubKeyBytes).toString('hex');
+        const k: any =
+            data.enclave_attested_application_public_key.claims.public_key.data
+        const pubKeyBytes: Uint8Array = ethers.decodeBase64(k)
+        const publicKeyHex: string = Buffer.from(pubKeyBytes).toString('hex');
 
-        const taBytes = ethers.decodeBase64(data.transitive_attested_function_call.transitive_attestation)
-        const ta = Buffer.from(taBytes).toString('utf-8');
+        const taBytes:Uint8Array = ethers.decodeBase64(
+            data.transitive_attested_function_call.transitive_attestation
+        )
+        const ta: string = Buffer.from(taBytes).toString('utf-8');
 
-        return {
-            publicKey: `0x${publicKeyHex}`,
-            transitiveAttestation: ta
-        };
+        return { publicKey: `0x${publicKeyHex}`, transitiveAttestation: ta};
     } catch (e) {
-        console.log(`e`, e)
+        throw new Error(`Error loading EVM link data: ` + e);
     }
 }
 
-const loadUserDeployedAddress = () => {
+const loadUserDeployedAddress: () => string  = () : string  => {
```

Comment:
Again, explicitly return type or throw error. Previously the implied return was `string | undefined` which threw errors below when passing into another function that only took `string` and didn't appreciate the potential for an `undefined` value

Reply:
Is there a reason why these functions are declared as const, while loadEVMLinkData is declared as function?